micrometer-core/src/test/java/io/micrometer/core/instrument/binder/httpcomponents/MicrometerHttpRequestExecutorTest.java

```diff
@@ -192,12 +213,120 @@ void overrideExtraTagsDoesNotThrowAnException(@WiremockResolver.Wiremock WireMoc
+                    @Override
+                    public String getName() {
+                        return "local." + super.getName();
+                    }
+                }).build();
+        HttpClient client = client(micrometerHttpRequestExecutor);
+        EntityUtils.consume(client.execute(new HttpGet(server.baseUrl())).getEntity());
+        assertThat(registry.get("local.custom.apache.http.client.requests")).isNotNull();
+    }
+
+    @Test
+    void localConventionConfigured(@WiremockResolver.Wiremock WireMockServer server) throws IOException {
+        server.stubFor(any(anyUrl()));
+        ObservationRegistry observationRegistry = createObservationRegistry();
+        MicrometerHttpRequestExecutor micrometerHttpRequestExecutor = MicrometerHttpRequestExecutor.builder(registry)
+                .observationRegistry(observationRegistry).observationConvention(new CustomGlobalApacheHttpConvention())
+                .build();
+        HttpClient client = client(micrometerHttpRequestExecutor);
+        EntityUtils.consume(client.execute(new HttpGet(server.baseUrl())).getEntity());
+        assertThat(registry.get("custom.apache.http.client.requests")).isNotNull();
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = { "get", "post", "custom" })
+    void contextualNameContainsRequestMethod(String method, @WiremockResolver.Wiremock WireMockServer server)
+            throws IOException {
+        server.stubFor(any(anyUrl()));
+        TestObservationRegistry observationRegistry = TestObservationRegistry.create();
+        MicrometerHttpRequestExecutor micrometerHttpRequestExecutor = MicrometerHttpRequestExecutor.builder(registry)
+                .observationRegistry(observationRegistry).build();
+        HttpClient client = client(micrometerHttpRequestExecutor);
+        if ("get".contentEquals(method)) {
+            EntityUtils.consume(client.execute(new HttpGet(server.baseUrl())).getEntity());
+        }
+        else if ("post".contentEquals(method)) {
+            EntityUtils.consume(client.execute(new HttpPost(server.baseUrl())).getEntity());
+        }
+        else {
+            EntityUtils.consume(client.execute(new HttpCustomMethod(method, server.baseUrl())).getEntity());
+        }
+        TestObservationRegistryAssert.assertThat(observationRegistry).hasSingleObservationThat()
+                .hasContextualNameEqualToIgnoringCase("http " + method);
+    }
+
+    private static class HttpCustomMethod extends HttpEntityEnclosingRequestBase {
+
+        private final String method;
+
+        private HttpCustomMethod(String method, String uri) {
+            super();
+            setURI(URI.create(uri));
+            this.method = method;
+        }
+
+        @Override
+        public String getMethod() {
+            return method;
+        }
+
+    }
+
+    // TODO add test for status = IO_ERROR case.
```

Comment:
I wanted to add a test for this, but I wasn't sure how. If someone knows how to test this scenario with the Apache HTTP client, let me know.  The scenarios are when the response returned from `execute` is `null` and when calling `execute` results in an `IOException`